tools/eslint-rules/buffer-constructor.js

```diff
@@ -0,0 +1,28 @@
+/**
+ * @fileoverview Require use of new Buffer constructor methods in lib
+ * @author James M Snell
+ */
+'use strict';
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+const msg = 'Use of the Buffer() constructor has been deprecated. ' +
+            'Please use either Buffer.alloc(), Buffer.allocUnsafe(), ' +
+            'or Buffer.from()';
+
+function test(context, node, which) {
+  if (node.type === which && node.callee.name === 'Buffer') {
```

Comment:
`node.callee.name` is always undefined. You mean `node.argument.callee.name`.  I've been unable to make this rule flag anything as is, although maybe you are trying to catch specific issues I haven't considered.